tests/test_auth.py

```diff
@@ -19,10 +19,9 @@ def test_kernel_channels(client, authenticated_user):
 @pytest.mark.parametrize("auth_mode", ("noauth", "token", "user"))
 def test_root_auth(auth_mode, client):
     response = client.get("/")
-    expected = 404
-    if auth_mode == "noauth":
-        expected = 200
+    expected = 200
     assert response.status_code == expected
+    assert response.headers["content-type"] == "text/html; charset=utf-8"
```

Comment:
It would be nice to check that for `noauth` we don't land on the login page, and for `token` and `user` we do. Maybe we can check that we get the login page's `index.html` for the later?

Reply:
That isn't very easy. Both redirect to an HTML page, so both return the same. The only solution that comes to my mind is to check the redirected URL, but I googled it, and I don't know how to do it in FastAPI.

Reply:
Ok, then I think we should [disable the `login` plugin](https://github.com/jupyter-server/fps/pull/30) for this test, and check that we have a `200` for `noauth`, and a `401` for `token` and `user`.